wasm/src/lib.rs

```diff
@@ -133,6 +135,44 @@ pub fn get_estimates(
     })
 }
 
+#[wasm_bindgen]
+pub fn get_circuit(
+    sources: Vec<js_sys::Array>,
+    targetProfile: &str,
+    operation: Option<IOperationInfo>,
+    language_features: Vec<String>,
+) -> Result<JsValue, String> {
+    let sources = get_source_map(sources, &None);
+    let target_profile = Profile::from_str(targetProfile).expect("invalid target profile");
+
+    let mut interpreter = interpret::Interpreter::new(
+        true,
+        sources,
+        PackageType::Exe,
+        target_profile.into(),
+        LanguageFeatures::from_iter(language_features),
+    )
+    .map_err(interpret_errors_into_vs_diagnostics_json)?;
+
+    let circuit = interpreter
+        .circuit(match operation {
+            Some(p) => {
+                let o: language_service::OperationInfo = p.into();
+                CircuitEntryPoint::Operation(o.operation)
+            }
+            None => CircuitEntryPoint::EntryPoint,
+        })
+        .map_err(interpret_errors_into_vs_diagnostics_json)?;
+
+    serde_wasm_bindgen::to_value(&circuit).map_err(|e| e.to_string())
+}
+
+#[allow(clippy::needless_pass_by_value)]
+fn interpret_errors_into_vs_diagnostics_json(errs: Vec<qsc::interpret::Error>) -> String {
+    serde_json::to_string(&interpret_errors_into_vs_diagnostics(&errs))
+        .expect("serializing errors to json should succeed (or should it?)")
```

Comment:
Should it? :-p

Reply:
Uh oh, forgot I left this in! Yes, it should succeed...